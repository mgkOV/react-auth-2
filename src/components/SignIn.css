.signin-card {
  width: 500px;
  margin: 150px auto;
  color: #fff;
  text-align: center;
  padding: 15px;
  height: auto !important;
}
