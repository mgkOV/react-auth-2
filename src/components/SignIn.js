import React, { Component } from 'react';
import { reset } from 'redux-form';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { signIn } from '../actions';
import SignInForm from './SignInForm';
import './SignIn.css';

export class SignIn extends Component {
  handleSubmit = value => {
    const { reset, signIn } = this.props;
    const { username, password } = value;
    signIn({ username, password });
    reset('signIn');
  };

  render() {
    const { authenticateUser } = this.props;

    const singInPage = (
      <div className="card medium indigo darken-1 signin-card ">
        <h2>Вход</h2>
        <div className="row">
          <SignInForm onSubmit={this.handleSubmit} />
        </div>
      </div>
    );

    return authenticateUser === true ? <Redirect to="/app" /> : singInPage;
  }
}

export default connect(
  ({ authenticateUser }) => ({
    authenticateUser
  }),
  { reset, signIn }
)(SignIn);
