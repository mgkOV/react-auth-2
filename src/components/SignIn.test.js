import React from 'react';
import { shallow } from 'enzyme';
import SignIn from './SignIn';

describe('SignIn', () => {
  const signIn = shallow(<SignIn />);

  it('should render properly', () => {
    expect(signIn).toMatchSnapshot();
  });
});
