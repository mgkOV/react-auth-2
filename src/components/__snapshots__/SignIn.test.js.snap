// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Вход
</h2>,
        <div
          className="row"
>
          <ReduxForm
                    onSubmit={[Function]}
          />
</div>,
      ],
      "className": "card medium indigo darken-1 signin-card ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Вход",
        },
        "ref": null,
        "rendered": "Вход",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ReduxForm
            onSubmit={[Function]}
/>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Вход
</h2>,
          <div
            className="row"
>
            <ReduxForm
                        onSubmit={[Function]}
            />
</div>,
        ],
        "className": "card medium indigo darken-1 signin-card ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Вход",
          },
          "ref": null,
          "rendered": "Вход",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ReduxForm
              onSubmit={[Function]}
/>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
